### PR TITLE
Add navigation and diagnostics docs to new format

### PR DESCRIPTION
**Pull request by IMS94**
$subject

### CODE REVIEW

File: en/docs/edit-the-code/code-navigation.md [hunk 1]
```diff
@@ -1 +1,19 @@
-# Code navigation
+# Code Navigation
```
Comment (praneesha) [inline en/docs/edit-the-code/code-navigation.md:1]:
```suggestion
# Code navigation
```

File: en/docs/edit-the-code/code-navigation.md [hunk 2]
```diff
@@ -1 +1,19 @@
...
+
+## Go to Definition 
+
+For a symbol, this feature will navigate you to the definition of the particular symbol. For example, when invoking the `Go To Definition` option from a function call expression, you will be navigated to the definition of the function. Apart from jumping to the definition, the peek definition will also be supported. The behavior will be the same not only for the constructs within the sources in the current project but also for external modules and standard libraries as well.
```
Comment (praneesha) [inline en/docs/edit-the-code/code-navigation.md:5]:
```suggestion
This feature navigates you to the definition of a particular symbol. For example, when you invoke the `Go To Definition` option on a function call expression, it navigates you to the definition of the function. Apart from that, the peek definition will also be supported. The behavior will be the same not only for the constructs within the sources in the current project but also for external modules and standard libraries as well.
```

File: en/docs/edit-the-code/code-navigation.md [hunk 3]
```diff
@@ -1 +1,19 @@
...
+
+<img src="https://github.com/wso2/ballerina-plugin-vscode/blob/main/resources/images/go-to-definition.gif?raw=true" width="100%" />
+
+## Find all References
```
Comment (praneesha) [inline en/docs/edit-the-code/code-navigation.md:9]:
```suggestion
## Find all references
```

File: en/docs/edit-the-code/code-navigation.md [hunk 4]
```diff
@@ -1 +1,19 @@
-# Code navigation
+# Code Navigation
+
+## Go to Definition
```
Comment (praneesha) [inline en/docs/edit-the-code/code-navigation.md:3]:
```suggestion
## Go to definition 
```

File: en/docs/edit-the-code/code-navigation.md [hunk 5]
```diff
@@ -1 +1,19 @@
...
+
+<img src="https://github.com/wso2/ballerina-plugin-vscode/blob/main/resources/images/go-to-definition.gif?raw=true" width="100%" />
+
+## Find all References
+
+Invoking the references on a symbol will prompt you with all the symbol references in the current project.
+
+<img src="https://github.com/wso2/ballerina-plugin-vscode/blob/main/resources/images/find-all-references.gif?raw=true" width="100%" />
+
+## Rename Symbols
```
Comment (praneesha) [inline en/docs/edit-the-code/code-navigation.md:15]:
```suggestion
## Rename symbols
```

**Review by praneesha — APPROVED** [review]

File: en/docs/edit-the-code/code-navigation.md [hunk 6]
```diff
@@ -1 +1,31 @@
 # Code navigation
+
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/code-navigation.md:2]: intro

File: en/docs/edit-the-code/code-navigation.md [hunk 7]
```diff
@@ -1 +1,31 @@
...
+## Go to definition 
+
+This feature navigates you to the definition of a particular symbol. For example, when you invoke the `Go To Definition` option on a function call expression, it navigates you to the definition of the function.
+
+![Go to definition support](/en/docs/img/edit-the-code/code-navigation/go-to-definition.gif?raw=true)
+
+## Go to symbol
+
+This feature allows you to navigate symbols inside a file.
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/code-navigation.md:11]: Shall we explain in words how we actually do this? As I understood from [1] and from the giff, we need to use the `@` symbol to do this?

[1] https://code.visualstudio.com/docs/editor/editingevolved#_go-to-symbol

File: en/docs/edit-the-code/code-navigation.md [hunk 8]
```diff
@@ -1 +1,31 @@
 # Code navigation
+
+## Go to definition 
+
+This feature navigates you to the definition of a particular symbol. For example, when you invoke the `Go To Definition` option on a function call expression, it navigates you to the definition of the function.
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/code-navigation.md:5]: `when you invoke the `Go To Definition` option on a function call expression` is this done by clicking the function call expression and selecting the **Go To Definition** from the list? 
The giff was pretty fast.

File: en/docs/edit-the-code/code-navigation.md [hunk 9]
```diff
@@ -1 +1,31 @@
...
+
+## Find all references
+
+Invoking the references on a symbol will prompt you with all the symbol references in the current project.
+
+![Find all references](/en/docs/img/edit-the-code/code-navigation/find-all-references.png?raw=true)
+
+## Peek
+
+This feature allows you to peek the definition or references of a given symbol.
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/code-navigation.md:23]: What do we mean by peek?

https://code.visualstudio.com/docs/editor/editingevolved#_peek

File: en/docs/edit-the-code/code-navigation.md [hunk 10]
```diff
@@ -1 +1,31 @@
...
+
+## Go to symbol
+
+This feature allows you to navigate symbols inside a file.
+
+![Go to symbol](/en/docs/img/edit-the-code/code-navigation/go-to-symbol.gif?raw=true)
+
+## Find all references
+
+Invoking the references on a symbol will prompt you with all the symbol references in the current project.
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/code-navigation.md:17]: Shall we explain this a bit more? how to do this really?

File: en/docs/edit-the-code/diagnostics.md [hunk 11]
```diff
@@ -1 +1,15 @@
...
+
+The diagnostics show you the syntax and semantic errors in the source code. Varieties of diagnostics such as warnings and errors will be shown. They will show up in multiple places:
+
+* In the Status Bar, there is a summary of all errors and warnings counts.
+* You can click on the summary or view the PROBLEMS panel with a list of all current errors.
+* If you open a file that has errors or warnings, they will be rendered inline with the text and in the overview ruler.
+
+![Diagnostics](/en/docs/img/edit-the-code/diagnostics/diagnostics.png?raw=true)
+
+## Quickfixes
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/diagnostics.md:11]:
```suggestion
## Quick fixes
```

File: en/docs/edit-the-code/diagnostics.md [hunk 12]
```diff
@@ -1 +1,15 @@
...
+
+For a selected set of diagnostics, you can see the quick fixes. For example, the `variable assignment is required` diagnostic will have 2 associated quick fixes to create a new variable and ignore the return value.
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/diagnostics.md:13]:
```suggestion
For certain diagnostics, the possible quick fixes are suggested as code actions. You can easily select and apply the required quick fix from the list. 

For example, the `variable assignment is required` diagnostic suggests two quick fixes as follows: 
- Create a new variable
- Ignore the return value
```

File: en/docs/edit-the-code/diagnostics.md [hunk 13]
```diff
@@ -1 +1,15 @@
 # Diagnostics
+
+The diagnostics show you the syntax and semantic errors in the source code. Varieties of diagnostics such as warnings and errors will be shown. They will show up in multiple places:
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/diagnostics.md:3]:
```suggestion
Diagnostics are the syntax and semantic errors that are identified in the source code. These may include a variety of errors and warnings. Diagnostics will show up in multiple places:
```

File: en/docs/edit-the-code/diagnostics.md [hunk 14]
```diff
@@ -1 +1,15 @@
...
+
+* In the Status Bar, there is a summary of all errors and warnings counts.
+* You can click on the summary or view the PROBLEMS panel with a list of all current errors.
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/diagnostics.md:6]:
```suggestion
* The summary view and the PROBLEMS panel lists all of the current errors.
```

File: en/docs/edit-the-code/diagnostics.md [hunk 15]
```diff
@@ -1 +1,15 @@
...
+* If you open a file that has errors or warnings, they will be rendered inline with the text and in the overview ruler.
```
Comment (nilminiwso2) [inline en/docs/edit-the-code/diagnostics.md:7]:
```suggestion
* Errors or warnings in a Ballerina file will be rendered inline with the text and in the overview ruler.
```

Comment (nilminiwso2) [reply en/docs/edit-the-code/diagnostics.md:7]: Curious how errors and warning are rendered inline. Are we referring to the light bulb?
Also, will see if we can avoid the term `Overview ruler`.

File: en/docs/edit-the-code/code-navigation.md (hunk 6 above)
Comment (nilminiwso2) [reply en/docs/edit-the-code/code-navigation.md:2]: A TW will add this.

File: en/docs/edit-the-code/code-navigation.md (hunk 7 above)
Comment (IMS94) [reply en/docs/edit-the-code/code-navigation.md:11]: This is a standard feature in VSCode. So the developers are usually familiar with the shortcut/approach of searching the symbols.

File: en/docs/edit-the-code/code-navigation.md (hunk 8 above)
Comment (IMS94) [reply en/docs/edit-the-code/code-navigation.md:5]: Same as above. VSCode users are familiar with how to go to the definition, `right click -> click 'Go to Definition'` or `CTRL/CMD+Click` shortcut.

File: en/docs/edit-the-code/diagnostics.md (hunk 15 above)
Comment (IMS94) [reply en/docs/edit-the-code/diagnostics.md:7]: Redered inline here means the `red/yellow color underline` you will see if there's an error in your code.
Screenshot from 2023-01-09 14-22-48

File: en/docs/edit-the-code/code-navigation.md (hunk 8 above)
Comment (nilminiwso2) [reply en/docs/edit-the-code/code-navigation.md:5]: A TW will add this by referring VS code docs

File: en/docs/edit-the-code/code-navigation.md (hunk 7 above)
Comment (nilminiwso2) [reply en/docs/edit-the-code/code-navigation.md:11]: A TW will enhance these by referring the VS Code docs

File: en/docs/edit-the-code/code-navigation.md (hunk 10 above)
Comment (nilminiwso2) [reply en/docs/edit-the-code/code-navigation.md:17]: A TW will add this by referring the VS Code docs

File: en/docs/edit-the-code/code-navigation.md (hunk 9 above)
Comment (nilminiwso2) [reply en/docs/edit-the-code/code-navigation.md:23]: A TW will address this by referring the VS code docs.

File: en/docs/edit-the-code/diagnostics.md (hunk 15 above)
Comment (nilminiwso2) [reply en/docs/edit-the-code/diagnostics.md]: ack

**Review by nilminiwso2 — APPROVED** [review]

**Pull request merged**